Source/Fluxor/DependencyInjection/FeatureStateInfo.cs

```diff
@@ -35,13 +35,17 @@ internal class FeatureStateInfo
 		private Func<object> CreateFactoryFromParameterlessConstructor(
 			FeatureStateAttribute featureStateAttribute)
 		{
-			ConstructorInfo constructor = StateType.GetConstructor(
+			if (StateType.IsValueType)
```

Comment:
What if the struct state has a constructor with parameters?

Reply:
It's really only intended for marker structs, as a lightweight alternative to using a class instance. Since it is a value type, it will always have a `default`. Since the calling function is specific to pararmeterless .ctors, I assumed this was fine. I'm adding a unit test to ensure.

Reply:
Actually, states can't be structs because Fluxor checks for a state change in the feature using object.ReferenceEquals, which in the case of value types always returns true.

Reply:
Ah, ok. I see.